### PR TITLE
Maternal health risk prediction using machine learning

### LINKED ISSUE #262: 💡[Feature]:  Maternal Health risk prediction using Machine learning
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Objective:
To develop a predictive model and a web application that can classify the risk level (high, medium, low) for pregnant women based on various health parameters. The project will involve data analysis, preprocessing, feature engineering, model training, evaluation, and deployment using a Streamlit web application.

Dataset:
The dataset obtained from Kaggle includes the following columns:

https://www.kaggle.com/datasets/pyuxbhatt/maternal-health-risk?select=Maternal+Health+Risk+Data+Set.csv

Age: Age of the pregnant woman
SystolicBP: Systolic blood pressure
DiastolicBP: Diastolic blood pressure
BS: Blood sugar level
BodyTemp: Body temperature
HeartRate: Heart rate
RiskLevel: Risk level classification (high risk, medium risk, low risk)

Steps Involved:

1) Data Analysis:

Exploratory Data Analysis (EDA):
Understand the distribution of each feature.
Identify any patterns or trends in the data.
Detect outliers and anomalies.
Visualize relationships between features and the target variable (RiskLevel).

2) Data Preprocessing:

Handling Missing Values:
Identify and handle missing data through imputation or removal.
Data Transformation:
Normalize or standardize numerical features.
Encode categorical variables if any.
Feature Selection:
Select relevant features that contribute most to the prediction.

3) Feature Engineering:

Creating New Features:
Derive new features that might help in improving the model's performance.
Feature Scaling:
Apply scaling techniques like Min-Max scaling or Standard scaling to bring all features to a similar range.

4)Model Training

5) Model Deployment

Integrate the trained model with the Streamlit app.

### Use Case

The Maternal Health Risk Prediction model can be used by healthcare providers in prenatal care facilities to assess the risk levels of pregnant women during routine check-ups. By inputting patient data such as age, blood pressure, blood sugar levels, body temperature, and heart rate into the Streamlit web application, healthcare professionals can quickly determine whether a pregnant woman is at high, medium, or low risk of complications. This real-time risk assessment tool can guide doctors in making timely and informed decisions about additional tests, treatments, or interventions to ensure the well-being of both the mother and the baby.

### Benefits

Implementing this predictive model and web application can significantly enhance prenatal care by providing early warnings of potential health risks. It enables proactive management of maternal health, reduces the likelihood of severe complications, and improves overall pregnancy outcomes. Healthcare providers can allocate resources more effectively, prioritize high-risk cases, and offer personalized care plans. Additionally, the ease of use and accessibility of the Streamlit app ensure that even facilities with limited technical infrastructure can benefit from advanced predictive analytics, ultimately leading to better maternal and fetal health.

### Add ScreenShots

none

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by RajKhanke**
work done :
1) models used : Xgboost (applied on final data) ,Gbm , random forest ,Knn
2)data preprocessing ,feature engineering , EDA and analysis
3) Streamlit based web application for taking input and risk prediction

Resolves #262 

Demo :

https://github.com/user-attachments/assets/f20288a9-187c-4fe3-99d5-ee76d981428b

### CODE REVIEW

**Comment by RajKhanke** [comment]
@Avdhesh-Varshney @TAHIR0110 please merge this pr along with level and label of gssoc'24. it will resolve the issue no 262

File: Maternal health Risk Prediction/webapp.mp4
Comment (Avdhesh-Varshney) [inline Maternal health Risk Prediction/webapp.mp4:1]: delete this video

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]

**Comment by RajKhanke** [comment]
okay

**Comment by RajKhanke** [comment]
@Avdhesh-Varshney @TAHIR0110 i have completed the required changes  , now please merge this PR along with the level and label of gssoc'24

**Comment by RajKhanke** [comment]
@TAHIR0110 @Avdhesh-Varshney @Atharv714  ?

**Comment by RajKhanke** [comment]
@Avdhesh-Varshney ?

**Comment by TAHIR0110** [comment]
@RajKhanke kindly delete the demo vid

**Comment by TAHIR0110** [comment]
@RajKhanke merged!

**Pull request merged**